### DISCUSSION TITLE
New syntax

### DISCUSSION THREAD

**Discussion by erothe** [epfl-scitas/spack-sdploy#44] [Ideas]
I will change the syntax of `spack-sdploy` commands. The idea is that `spack-sdploy` commands use the same syntax as `spack` commands. This will make the use of `spack-sdploy` more natural from a `spack` point of view and less invasive to the user as the commands provided by the extension will likely be seen as `spack` commands to the user.

In practice, this proposal will change the commands like `spack write-config-yaml` to `spack write config`. The same goes on for every other command - no more dashes.

The whole change would be like:
```
clone-external-repos    clone <some-repo>
list-compilers          list compilers
list-mirrors            list mirrors
write-activate-list     write activate
write-concretizer-yaml  write concretizer
write-config-yaml       write config
write-mirrors-yaml      write mirror
write-modules-yaml      write modules
write-packages-yaml     write packages
write-repos-yaml        write repos
write-spack-yaml        write manifest
```

Things to discuss:

- instead of using the word `write`, we can use `deploy` as this is only related to spack configuration files - to deploy a configuration. In that case, `write activate` would remain `write activate`.

- it is a good idea to add a starter command - a command that will proceed every other command provided by this extension. This could be `sdploy` or `scitas`. The command `spack write config` would then become `spack scitas write config` or `spack sdploy write config`. This will allow for a better separation between what are `spack` commands and what are `spack-sdploy` commands. This will also prevent conflicts with a future `spack` command if tomorrow `spack` will add its own `spack write` ou `spack deploy` (a command the extension already provides).

Your comments are welcome.

**Comment by erothe** [comment]
I propose:
```
spack scitas clone <some-repo>
spack scitas list compilers
spack scitas list mirrors
spack scitas write activate
spack scitas deploy concretizer
spack scitas deploy config
spack scitas deploy mirror
spack scitas deploy modules
spack scitas deploy packages
spack scitas deploy repos
spack scitas deploy manifest
```

**Reply by batatasfritas** [reply]
I like this option. I agree that the starter command makes it more explicit and if we think about the having autocomplete on the commands it reduces the subset of options to "explore" in case the user is not fully accustomed to the commands yet. And it's unlikely scitas will be used by Spack centrally so it's a good option.
